File: Two Pointers/Javascript/Valid_Palindrome_simplified-claude.js
const s = "racecar";

function is_palindrom() {
  let left = 0;
  let right = s.length - 1;
  while (left < right) {
    if (s[left] != s[right]) {
      console.log("false");
      return false;
    }
    left += 1;
    right -= 1;
  }
  console.log("true");
  return true;
}

is_palindrom()

File: Two Pointers/Python/Valid_Palindrome_simplified-claude.py
# Given a string s, return true if it reads the same forwards and backwards.
# Examples :- "racecar" → true, "hello" → false, "a" → true


# Dont do this on Interview
# s = "hello"
# l = s[::-1]
# if s == l:
#     print("true")
# else:
#     print("false")


s = "racecar"


def is_palindrom():
    left = 0
    right = len(s) - 1
    while left < right:
        if s[left] != s[right]:
            print("false")
            return False
        left += 1
        right -= 1
    print("true")
    return True

is_palindrom()
